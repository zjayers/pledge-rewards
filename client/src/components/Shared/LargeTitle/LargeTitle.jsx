import React from 'react';
import LargeTitle from './LargeTitle.style';

export default (props) => {
  const { uppercase, children } = props;
  return <LargeTitle uppercase={uppercase}>{children}</LargeTitle>;
};
