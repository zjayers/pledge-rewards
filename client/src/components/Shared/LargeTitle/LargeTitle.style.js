import styled from 'styled-components';

export default styled.h3`
  font-size: 2.1rem;
  color: ${(props) => props.theme.colors.text};
  text-transform: ${(props) => (props.uppercase ? 'uppercase' : 'capitalize')};
  font-weight: 400;
  line-height: 1.4;
  margin: 4rem 0;
`;
