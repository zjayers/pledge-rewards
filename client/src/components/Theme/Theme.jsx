import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
