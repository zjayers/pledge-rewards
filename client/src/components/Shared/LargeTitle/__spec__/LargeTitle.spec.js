import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import LargeTitle from '../LargeTitle';
import Theme from '../../../Theme/Theme';

describe('LargeTitle Component', () => {
  let component;
  let componentLowercase;

  beforeEach(() => {
    component = mount(
      <Theme>
        <LargeTitle uppercase />
      </Theme>
    );

    componentLowercase = mount(
      <Theme>
        <LargeTitle uppercase={false} />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should match the test snapshot for lowercase lettering', () => {
    expect(toJson(componentLowercase)).toMatchSnapshot();
  });
});
