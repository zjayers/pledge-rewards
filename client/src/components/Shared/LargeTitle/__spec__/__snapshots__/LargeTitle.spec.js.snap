// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeTitle Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default
      uppercase={true}
    >
      <LargeTitlestyle___default
        uppercase={true}
      >
        <h3
          className="LargeTitlestyle___default-sc-68uopl-0 kTxtyv"
        />
      </LargeTitlestyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;

exports[`LargeTitle Component should match the test snapshot for lowercase lettering 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default
      uppercase={false}
    >
      <LargeTitlestyle___default
        uppercase={false}
      >
        <h3
          className="LargeTitlestyle___default-sc-68uopl-0 iORtcV"
        />
      </LargeTitlestyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;
